integration-test/main.go

```diff
@@ -822,6 +842,289 @@ func testDLQFlow() {
 	}
 }
 
+func testScheduledMessages() {
+	log.Println("Starting scheduled messages test")
+	
+	// 1. Test basic scheduled message flow
+	testBasicScheduledMessageFlow()
+	
+	// 2. Test error cases
+	testScheduledMessageErrorCases()
+	
+	// 3. Test concurrent scheduling
+	testConcurrentScheduledMessages()
+	
+	log.Println("Scheduled messages tests completed successfully")
+}
+
+func testBasicScheduledMessageFlow() {
+	log.Println("Testing basic scheduled message flow")
+	channelID := generalChannelID
+	
+	// Schedule a message for delivery in 3 minutes (must be at least 2 minutes in the future)
+	futureTime := time.Now().Add(3 * time.Minute).Format(time.RFC3339)
+	
+	// Create a scheduled message
+	req, _ := http.NewRequest(http.MethodPost, brokerBaseURL+"/channel/"+channelID+"/broadcast", strings.NewReader(payload))
+	req.Header.Add(headerContentType, contentType)
+	req.Header.Add(headerChannelToken, token)
+	req.Header.Add(headerProducerID, producerID)
+	req.Header.Add(headerProducerToken, token)
+	req.Header.Add(headerScheduledFor, futureTime)
+	
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Println("Error creating scheduled message:", err)
+		os.Exit(40)
+	}
+	defer resp.Body.Close()
+	
+	if resp.StatusCode != http.StatusCreated {
+		body, _ := io.ReadAll(resp.Body)
+		log.Println("Error scheduling message:", resp.StatusCode, string(body))
+		os.Exit(41)
+	}
+	
+	// Get message URL from the Location header
+	messageURL := resp.Header.Get("Location")
+	if messageURL == "" {
+		log.Println("No location header in response")
+		os.Exit(42)
+	}
+	
+	log.Println("Created scheduled message:", messageURL)
+	
+	// Get scheduled message list to verify it appears
+	scheduledMessagesURL := brokerBaseURL + "/channel/" + channelID + "/scheduled-messages"
+	req, _ = http.NewRequest(http.MethodGet, scheduledMessagesURL, nil)
+	resp, err = client.Do(req)
+	if err != nil {
+		log.Println("Error getting scheduled messages:", err)
+		os.Exit(43)
+	}
+	defer resp.Body.Close()
+	
+	body, _ := io.ReadAll(resp.Body)
+	var msgList ScheduledMessageListResult
+	err = json.Unmarshal(body, &msgList)
+	if err != nil {
+		log.Println("Error parsing scheduled messages response:", err)
+		os.Exit(44)
+	}
+	
+	if len(msgList.Result) < 1 {
+		log.Println("No scheduled messages found")
+		os.Exit(45)
+	}
+	
+	// Verify message status endpoint contains scheduled message count
+	statusURL := brokerBaseURL + "/channel/" + channelID + "/messages-status"
+	req, _ = http.NewRequest(http.MethodGet, statusURL, nil)
+	resp, err = client.Do(req)
+	if err != nil {
+		log.Println("Error getting messages status:", err)
+		os.Exit(46)
+	}
+	defer resp.Body.Close()
+	
+	body, _ = io.ReadAll(resp.Body)
```

Comment:
Are we missing an assertion here on unmarshalled `body`?

Reply:
Added assertion below.